pkg/cache/util.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+// WeightAllocator is used to alloc weight for cache.
+// When we need to use multiple items from the cache,
+// we may need to assign different weights to these items.
+// WeightAllocator will divide these items into some segments whose number named as segNum which should great than 0.
+// And the items at first segment will be assigned more weight that is `segNum` times that of item at last segment.
+// If you want assign same weights, just input segNum as 1.
+// If length is 10 and segNum is 3, the weight is [0.143,0.143,0.143,0.143,0.095,0.095,0.095,0.047,0.047,0.047];
```

Comment:
you can describe this sequence  as  arr [3,3,3,3,2,2,2,1,1,1],  and then uniform them :  arr[3,3,3,3,2,2,2,1,1,1]/sum(arr)=arr/21